src/lib/protocols/present-proof/messages/PresentationMessage.ts

```diff
@@ -0,0 +1,68 @@
+import { Equals, IsArray, IsString, ValidateNested, IsOptional } from 'class-validator';
+import { Expose, Type } from 'class-transformer';
+
+import { AgentMessage } from '../../../agent/AgentMessage';
+import { Attachment } from '../../../decorators/attachment/Attachment';
+import { JsonEncoder } from '../../../utils/JsonEncoder';
+import { PresentProofMessageType } from './PresentProofMessageType';
+
+export const INDY_PROOF_ATTACHMENT_ID = 'libindy-presentation-0';
+
+export interface PresentationOptions {
+  id?: string;
+  comment?: string;
+  attachments: Attachment[];
+}
+
+/**
+ * Presentation Message part of Present Proof Protocol used as a response to a {@link PresentationRequestMessage | Presentation Request Message} from prover to verifier.
+ * Contains signed presentations.
+ *
+ * @see https://github.com/hyperledger/aries-rfcs/blob/master/features/0037-present-proof/README.md#presentation
+ */
+export class PresentationMessage extends AgentMessage {
+  public constructor(options: PresentationOptions) {
+    super();
+
+    if (options) {
+      this.id = options.id ?? this.generateId();
+      this.comment = options.comment;
+      this.attachments = options.attachments;
+    }
+  }
+
+  @Equals(PresentationMessage.type)
+  public readonly type = PresentationMessage.type;
+  public static readonly type = PresentProofMessageType.Presentation;
+
+  /**
+   *  Provides some human readable information about this request for a presentation.
+   */
+  @IsOptional()
+  @IsString()
+  public comment?: string;
+
+  /**
+   * An array of attachments containing the presentation in the requested format(s).
+   */
+  @Expose({ name: 'request-presentations~attach' })
+  @Type(() => Attachment)
+  @IsArray()
+  @ValidateNested({
+    each: true,
+  })
+  public attachments!: Attachment[];
+
+  public get indyProof(): IndyProof | null {
```

Comment:
This method introduces a dependency on Indy.  Not sure how to work around that, just saying 😄

Reply:
Not necessarily. It creates a dependency on indy types. But with my new PR (#183) the compiled JS will have no reference to Indy at all. It's only for type context.

And as the v1 protocols are very indy specific I don't see the problem in adding indy specific utility methods. In v2 we'll do it different of course